python_modules/dagster/dagster/_core/definitions/freshness_checks/non_partitioned.py

```diff
@@ -0,0 +1,160 @@
+from datetime import datetime
+from typing import Optional, Sequence, Union
+
+import pendulum
+
+from dagster import _check as check
+from dagster._core.definitions.asset_check_spec import AssetCheckSeverity
+from dagster._core.definitions.metadata import (
+    MetadataValue,
+    TimestampMetadataValue,
+)
+from dagster._core.execution.context.compute import AssetExecutionContext
+from dagster._utils.schedules import is_valid_cron_string, reverse_cron_string_iterator
+
+from ..asset_check_result import AssetCheckResult
+from ..asset_checks import AssetChecksDefinition
+from ..assets import AssetsDefinition, SourceAsset
+from ..events import CoercibleToAssetKey
+from .utils import (
+    DEFAULT_FRESHNESS_SEVERITY,
+    asset_to_keys_iterable,
+    ensure_no_duplicate_assets,
+    get_last_updated_timestamp,
+    retrieve_latest_record,
+)
+
+
+def build_non_partitioned_freshness_checks(
+    *,
+    assets: Sequence[Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset]],
+    freshness_cron: Optional[str] = None,
+    maximum_lag_minutes: Optional[int] = None,
```

Comment:
Should `maximum_lag_minutes` be required? Having trouble conceptualizing how this works without it.

Reply:
I buy being maximally restrictive with the parameterizations. I did add a test, but I agree it's kinda weird.